lib/fs.js

```diff
@@ -338,33 +342,45 @@ function readFileAfterOpen(err, fd) {
   }
 
   context.fd = fd;
-
-  var req = new FSReqWrap();
-  req.oncomplete = readFileAfterStat;
-  req.context = context;
-  binding.fstat(fd, req);
+  // On Windows fstat cannot be called on std fds, so just use 0 as its size,
+  // which would result in a list of buffers to be concatenated
+  if (isStdFD(fd)) {
```

Comment:
If it's just Windows, should we be checking `process.platform === 'win32' && isStdFD(fd)` instead?

Reply:
@mscdex hmm, yes, it would probably favor the case on non-windows platforms where the stdin is redirected from a regular file.

Reply:
Or, the check probably is unnecessary since on Windows `fstat` still works on stdin if it's redirected. The `isUserFd` should already be enough in case of errors.